@xen-orchestra/lite/src/components/vm/network/VmVifsTable.vue

```diff
@@ -0,0 +1,290 @@
+        <div>{{ $t('no-result') }}</div>
+      </VtsStateHero>
+      <UiTopBottomTable :selected-items="0" :total-items="0" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import useMultiSelect from '@/composables/multi-select.composable'
+import type { XenApiNetwork, XenApiVif } from '@/libs/xen-api/xen-api.types'
+import { useNetworkStore } from '@/stores/xen-api/network.store'
+import { useVifStore } from '@/stores/xen-api/vif.store'
+import { useVmGuestMetricsStore } from '@/stores/xen-api/vm-guest-metrics.store'
+import { useVmStore } from '@/stores/xen-api/vm.store'
+import VtsConnectionStatus from '@core/components/connection-status/VtsConnectionStatus.vue'
+import VtsDataTable from '@core/components/data-table/VtsDataTable.vue'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import VtsStateHero from '@core/components/state-hero/VtsStateHero.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTableActions from '@core/components/ui/table-actions/UiTableActions.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { useRouteQuery } from '@core/composables/route-query.composable'
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import {
+  faAlignLeft,
+  faAt,
+  faCaretDown,
+  faEdit,
+  faEllipsis,
+  faHashtag,
+  faPlus,
+  faPowerOff,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { noop } from '@vueuse/shared'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { vifs } = defineProps<{
+  vifs: XenApiVif[]
+}>()
+
+const { isReady, hasError } = useVifStore().subscribe()
+const { getByOpaqueRef: getNetworkByOpaqueRef } = useNetworkStore().subscribe()
+const { getByOpaqueRef: getGuestMetricsByOpaqueRef } = useVmGuestMetricsStore().subscribe()
+const { getByOpaqueRef: getVmByOpaqueRef } = useVmStore().subscribe()
+
+const { t } = useI18n()
+
+const selectedVifId = useRouteQuery('id')
+
+const vifsUuids = computed(() => vifs.map(vif => vif.uuid))
+
+const { selected, areAllSelected } = useMultiSelect(vifsUuids)
+
+const getNetworkName = (networkRef: XenApiNetwork['$ref']) => getNetworkByOpaqueRef(networkRef)?.name_label
+
+const getIpAddresses = (vif: XenApiVif) => {
+  const vm = getVmByOpaqueRef(vif.VM)
+
+  if (vm) {
+    const networks = getGuestMetricsByOpaqueRef(vm.guest_metrics)?.networks
+
+    if (networks) {
+      return [...new Set(Object.values(networks).sort())]
+    }
+  }
+  return []
+}
```

Comment:
Maybe you can do something with early returns ? 

```suggestion
const getIpAddresses = (vif: XenApiVif) => {
  const vm = getVmByOpaqueRef(vif.VM)
  if (!vm) return []

  const guestMetrics = getGuestMetricsByOpaqueRef(vm.guest_metrics)
  if (!guestMetrics?.networks) return []

  return [...new Set(Object.values(guestMetrics.networks).sort())]
}
```